lib/internal/Magento/Framework/View/Asset/Repository.php

```diff
@@ -146,7 +146,7 @@ public function updateDesignParams(array &$params)
         }
 
         if ($theme) {
-            $params['themeModel'] = $this->getThemeProvider()->getThemeByFullPath($area . '/' . $theme);
+            $params['themeModel'] = $this->getThemeProvider()->getThemeById($theme);
```

Comment:
Actually, also for line 145 `getThemeById` should be called as `getConfigurationDesignTheme` returns the id. I don't have time to write integration test at this moment, unfortunately.
Should I open another PR on develop branch?